var/spack/repos/builtin/packages/otf-cpt/package.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class OtfCpt(CMakePackage):
+    """Tool to collect and report model factors (aka. fundamental performance factors)
+    for hybrid MPI + OpenMP applications on-the-fly."""
+
+    # Add a proper url for your package's homepage here.
+    homepage = (
+        "https://github.com/RWTH-HPC/OTF-CPT?tab=readme-ov-file#on-the-fly-critical-path-tool"
+    )
+    git = "https://github.com/RWTH-HPC/OTF-CPT.git"
+
+    maintainers("jgraciahlrs", "jprotze")
+
+    license("Apache-2.0", checked_by="jgraciahlrs")
+
+    version("0.9", tag="v0.9")
+
+    depends_on("cxx", type="build")
+    depends_on("mpi")
+    conflicts(
+        "%gcc",
+        msg="GCC compilers currently do not support OMPT, please use a clang-like compiler.",
```

Comment:
@jgraciahlrs / @jprotze: I ran into the issue where you need to have the `libomp` for the chosen clang version installed as well, and would like to amend the message to include it:


```suggestion
        # Use a clang compiler with a matching libomp, e.g. 'sudo apt install libomp-14-dev':
        msg="gcc currently does not support OMPT, please use a clang-like compiler with libomp",
```
I tried to shorten the first part of the sentence to keep the line length within the 100 column limit.

PS: This was the error I got, which was fixed by `sudo apt install libomp-14-dev`:
```py
==> otf-cpt: Executing phase: 'cmake'
-- The CXX compiler identification is Clang 14.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: lib/spack/env/clang/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: lib/spack/env/clang/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found MPI_C: opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openmpi-5.0.5-hitv2aowpoesno5qi3u2uiwlxtej4qw2/lib/libmpi.so (found version "3.1")
-- Found MPI_CXX: opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openmpi-5.0.5-hitv2aowpoesno5qi3u2uiwlxtej4qw2/lib/libmpi.so (found version "3.1")
-- Found MPI: TRUE (found version "3.1")
CMake Error at opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/cmake-3.30.5-bz2fjg27di5kg7b5bpielyt3xnw7axrc/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
Call Stack (most recent call first):
  opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/cmake-3.30.5-bz2fjg27di5kg7b5bpielyt3xnw7axrc/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/cmake-3.30.5-bz2fjg27di5kg7b5bpielyt3xnw7axrc/share/cmake-3.30/Modules/FindOpenMP.cmake:600 (find_package_handle_standard_args)
  CMakeLists.txt:11 (find_package)
```